config-spec.md

```diff
@@ -65,3 +67,76 @@ workflow:
+
+# Config Spec v0.0.2
+
+<details>
+
+<summary>details</summary>
+
+## Key concepts
+
+- **signals** - This is a global state store that can be accessed by all parts of the workflow.
+  - This will allow us to easily share state across workflows, components, and data.
+  - Workflows, components, and data declare what signals they need and we can validate at config compile time when the workflow is not configured with the needed signals. If a signal is is missing we can do things like show a warning to the user or auto-suggest adding signals to their config.
+- **frameRender** - This is tells the render how to render the current level of frames.
+  - By default it will render frames inline.
+  - The `sequence` annotation tells the renderer to render only one frame at a time. We will pass in navigation functions into components which will allow them to tell the render to move to the next / prev frames. We can persist a cursor to local / server storage saying where the user is in the workflow and the renderer can use that to render the appropriate frame on page load.
+- **storedQuery** - This data type tells the renderer to use stored queries to fetch results.
+  - If the current url matches `/stored_queries/id/sessions/id`
+    - it issues a GET request to fetch results from that url (it shows a loading indicator while results are being fetched by the stored query backend)
+    - else if there are signals in the global store that match the parameters needed for the query
+      - It sends a POST to create stored query, sets the cursor for the current location in the workflow, then redirects to the stored query url returned by the POST. The redirect will trigger the first part of this condition.
+    - else redirect the user back to the beginning of the workflow.
+  - The `pageSize` annotation sets how many results to show per page. Default 1.
+  - Components will be given navigation functions. There will be a navigation function that moves to the next page of results (e.g. increments the pagination, fetches more results, and re-renders the frame).
+- **Local / server store** - We can persist values to local storage or the server.
+  - This is useful for persisting workflow and pagination cursors among other values.
+  - For the server store, we can add a `store` column to the workflow sessions table to hold arbitrary json data.
+
+## Sample configuration
+
+```yaml
+# Worfklow name: Get all issues for a given user
+# This is a multi-stage workflow
+# 1. Show input form to add account_id
+# 2. User inputs account_id
+# 3. Shows list of issues belonging to that account_id
+workflow:
+  version: 1
+  signals:
+    - account_id: number
+  renderFrames: InSequence
+  frames:
+    - frame:
+      view:
+        component: OTInputForm
+        props:
+          fields:
+            - :account_id
+    - frame:
+      data:
+        query: >\n
+          SELECT id AS issue_id
+          FROM issues
+          WHERE author_id = :account_id
+        params:
```

Comment:
Per our conversation, we're doing to just declare parameters here as opposed to trying to extract them from the query. Furthermore, we don't have any type information in the query and couldn't do this anyway.

Pie in the sky idea 💡. In the distant future when we support integration with data catalogs, I wonder if we could get type information from that and apply it to the query? I feel like having queries dynamically typed would be a great feature. Another way we could do this too perhaps would be to run explain queries and extract type information that way.